File: frontend/src/types/video.ts
export interface KlingSettings {
  model_name: string        // "kling-v1.6" | "kling-v2.0" | "kling-v2.6"
  mode: string              // "std" | "pro"
  duration: string          // "5" | "10"
  aspect_ratio: string      // "16:9" | "9:16" | "1:1"
  cfg_scale: number
  negative_prompt?: string
  sound?: string            // "on" | "off"
  voice_list?: string[]
}

export interface KlingPreset {
  name: string
  settings: KlingSettings
}

export interface VideoGenerateRequest {
  image_path: string
  prompt?: string
  kling_settings: KlingSettings
  batch_id?: string
}

export interface VideoBatchItem {
  image_path: string
  prompt?: string
  variation_count: number
}

export interface VideoBatchRequest {
  items: VideoBatchItem[]
  kling_settings: KlingSettings
}

export interface VideoItem {
  id: number
  execution_id: string
  filename?: string
  source_image?: string
  prompt: string
  status: string
  created_at: string
  batch_id?: string
  video_url?: string
  thumbnail_url?: string
}

export interface VideoListResponse {
  items: VideoItem[]
  total: number
  page: number
  pages: number
}

export interface VideoStatusResponse {
  task_id: string
  status: string   // "pending" | "processing" | "succeed" | "failed"
  progress: number
  video_url?: string
  local_path?: string
  duration?: string
}

export interface VideoMergeRequest {
  filenames: string[]
  transition_type: string   // "Crossfade" | "Fade to Black" | "Simple Cut"
  transition_duration: number
}

export interface StoryboardVariation {
  variation: number
  concept_name: string
  prompt: string
}

export interface StoryboardResult {
  source_image: string
  persona: string
  variations: StoryboardVariation[]
}

export interface StoryboardResponse {
  results: StoryboardResult[]
}

export interface MusicAnalysisResponse {
  vibe: string
  lyrics: string
  analysis?: string
}

File: backend/models/video.py
from typing import List, Optional
from pydantic import BaseModel, Field


class KlingSettings(BaseModel):
    model_name: str = "kling-v1.6"
    mode: str = "std"  # "std" or "pro"
    duration: str = "5"  # "5" or "10"
    aspect_ratio: str = "16:9"
    cfg_scale: float = Field(default=0.5, ge=0.0, le=1.0)
    negative_prompt: Optional[str] = None
    sound: Optional[str] = None  # "on" or "off"
    voice_list: Optional[List[str]] = None


class VideoGenerateRequest(BaseModel):
    image_path: str
    prompt: Optional[str] = None
    kling_settings: KlingSettings = Field(default_factory=KlingSettings)
    batch_id: Optional[str] = None


class VideoGenerateResponse(BaseModel):
    task_id: str
    batch_id: Optional[str] = None
    status: str = "pending"


class VideoBatchItem(BaseModel):
    image_path: str
    prompt: Optional[str] = None
    variation_count: int = Field(default=1, ge=1, le=5)


class VideoBatchRequest(BaseModel):
    items: List[VideoBatchItem]
    kling_settings: KlingSettings = Field(default_factory=KlingSettings)


class VideoBatchResponse(BaseModel):
    batch_id: str
    task_ids: List[str]


class VideoStatusResponse(BaseModel):
    task_id: str
    status: str
    progress: int = 0
    video_url: Optional[str] = None
    local_path: Optional[str] = None
    duration: Optional[str] = None


class VideoItem(BaseModel):
    id: int
    execution_id: str
    filename: Optional[str] = None
    source_image: Optional[str] = None
    prompt: str
    status: str
    created_at: str
    batch_id: Optional[str] = None
    video_url: Optional[str] = None
    thumbnail_url: Optional[str] = None


class VideoListResponse(BaseModel):
    items: List[VideoItem]
    total: int
    page: int
    pages: int


class VideoMergeRequest(BaseModel):
    filenames: List[str]
    transition_type: str = "Crossfade"
    transition_duration: float = Field(default=0.5, ge=0.0, le=2.0)


class VideoMergeResponse(BaseModel):
    task_id: str
    output_filename: str


class MusicAnalysisRequest(BaseModel):
    audio_path: str


class MusicAnalysisResponse(BaseModel):
    vibe: str
    lyrics: str
    analysis: Optional[str] = None


class StoryboardRequest(BaseModel):
    image_paths: List[str]
    vision_model: str = "gpt-4o"
    persona: str = "default"
    variation_count: int = Field(default=3, ge=1, le=5)


class StoryboardVariation(BaseModel):
    variation: int
    concept_name: str
    prompt: str


class StoryboardResult(BaseModel):
    source_image: str
    persona: str
    variations: List[StoryboardVariation]


class StoryboardResponse(BaseModel):
    results: List[StoryboardResult]


class KlingPreset(BaseModel):
    name: str
    settings: KlingSettings
